### PR TITLE
Fix multi-extension builds

### PR DESCRIPTION
**Pull request by isaacroldan**
### WHY are these changes introduced?
With Unified config you can define multiple ui-extensions in a single folder.
But our build system wasn't ready for that as the output path was the same for all extensions in a given folder, which caused that the last extension being built overrides all previous builds.

### WHAT is this pull request doing?
- Each extension is now built using the handle as the outputFileName, that way we'll have multiple files inside `dist`, one per extension.
- Update the dev-server to serve files using this new format

### How to test your changes?
- Defining an admin action and block extension in the same folder should work for `dev`

### Measuring impact

How do we know this change was effective? Please choose one:

- [ ] n/a - this doesn't need measurement, e.g. a linting rule or a bug-fix
- [ ] Existing analytics will cater for this addition
- [ ] PR includes analytics changes to measure impact

### Checklist

- [ ] I've considered possible cross-platform impacts (Mac, Linux, Windows)
- [ ] I've considered possible [documentation](https://shopify.dev) changes
- [ ] I've made sure that any changes to `dev` or `deploy` have been reflected in the [internal flowchart](https://www.figma.com/file/7vqUp50u6dm48Zfb4JRRn8/CLI3-Internals).

### CODE REVIEW

File: packages/app/src/cli/models/extensions/extension-instance.ts [hunk 1]
```diff
@@ -99,6 +99,10 @@ export class ExtensionInstance<TConfiguration extends BaseConfigType = BaseConfi
     return this.specification.appModuleFeatures(this.configuration)
   }
 
+  get outputFileName() {
+    return `${this.handle}.js`
```
[the remark this answers is not in the document]
Comment (isaacroldan) [reply packages/app/src/cli/models/extensions/extension-instance.ts:103]: This handle can never be `undefined`
It is mandatory for extensions in the UnifiedSchema and for the legacy ones we take the `directoryName` as fallback handle

[the remark this answers is not in the document]
Comment (davejcameron) [reply packages/app/src/cli/models/extensions/extension-instance.ts:103]: It's actually the `configuration.name` as fallback slugified

[the remark this answers is not in the document]
Comment (isaacroldan) [reply packages/app/src/cli/models/extensions/extension-instance.ts:103]: you're right!

[the remark this answers is not in the document]
Comment (developit) [reply packages/app/src/cli/models/extensions/extension-instance.ts:103]: Are you going to be updating this PR with the handle fallback calculation, or was that already implemented?

File: packages/app/src/cli/services/dev/extension/bundler.test.ts [hunk 2]
```diff
@@ -109,7 +109,7 @@ describe('setupBundlerAndFileWatcher()', () => {
     expect(bundle.bundleExtension).toHaveBeenCalledWith(
       expect.objectContaining({
         minify: false,
-        outputPath: 'directory/path/2/dist/main.js',
+        outputPath: 'directory/path/2/dist/test-ui-extension.js',
```
[the remark this answers is not in the document]
Comment (isaacroldan) [reply packages/app/src/cli/services/dev/extension/bundler.test.ts]: Make it more explicit in the tests 👌

**Pull request merged**